stdlib/public/core/SmallString.swift

```diff
@@ -241,20 +241,36 @@ extension _SmallString {
   fileprivate mutating func withMutableCapacity(
     _ f: (UnsafeMutableRawBufferPointer) throws -> Int
   ) rethrows {
-    let len = try withUnsafeMutableBytes(of: &self._storage) {
+    var tmp = self._storage
+    let len = try withUnsafeMutableBytes(of: &tmp) {
       (rawBufPtr: UnsafeMutableRawBufferPointer) -> Int in
-      let len = try f(rawBufPtr)
-      UnsafeMutableRawBufferPointer(
-        rebasing: rawBufPtr[len...]
-      ).initializeMemory(as: UInt8.self, repeating: 0)
+      var len = try f(rawBufPtr)
+      
+      if len == 0 {
+        return 0
+      }
+      
+      _internalInvariant(len <= _SmallString.capacity)
+      
+      self._storage = (0, 0)
+      withUnsafeMutableBytes(of: &self._storage) { storagePtr in
+        var i = 0
+        while i < len {
+          storagePtr.storeBytes(
+            of: rawBufPtr.load(fromByteOffset: i, as: UInt8.self),
```

Comment:
`toByteOffset: i`

Reply:
doh! I knew I was going to miss something doing it this way